`Could not find a declaration file for module` when using `moduleResolution: nodenext` in Typescript

**What package within Headless UI are you using?**

All packages.

**What version of that package are you using?**

v0.1.2

**What browser are you using?**

N/A

**Reproduction URL**

[Link to Codesandbox reproduction environment](https://codesandbox.io/p/sandbox/romantic-field-8stf5y?file=%2FREADME.md).
Open a new terminal and run `yarn build`, it should output the following error:

```
yarn run v1.22.19
$ tsc
index.ts:1:20 - error TS7016: Could not find a declaration file for module '@headlessui/tailwindcss'. '/project/home/luudjanssen/workspace/node_modules/@headlessui/tailwindcss/dist/index.cjs' implicitly has an 'any' type.
  Try `npm i --save-dev @types/headlessui__tailwindcss` if it exists or add a new declaration (.d.ts) file containing `declare module '@headlessui/tailwindcss';`

1 import Plugin from "@headlessui/tailwindcss";
                     ~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in index.ts:1

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Describe your issue**

The Typescript (/ Node.js) module resolution algorithm that supports the `exports` field in `package.json` expects paths in the `exports` declaration to also contain a link to the typescript types:

```json
{
  "exports": {
    "require": "./dist/index.cjs",
    "types": "./dist/index.d.ts"
  },
}
```

The `types` field is missing in all of the `package.json` files in `headlessui`. 

> ⚠️ I opened this issue with the intent to fix it in a PR.

--- a/packages/@headlessui-tailwindcss/package.json
+++ b/packages/@headlessui-tailwindcss/package.json
@@ -11,9 +11,9 @@
     "dist"
   ],
   "exports": {
+    "types": "./dist/index.d.ts",
     "import": "./dist/headlessui.esm.js",
-    "require": "./dist/index.cjs",
-    "types": "./dist/index.d.ts"
+    "require": "./dist/index.cjs"
   },
   "sideEffects": false,
   "engines": {

--- a/packages/@headlessui-tailwindcss/package.json
+++ b/packages/@headlessui-tailwindcss/package.json
@@ -4,13 +4,16 @@
   "description": "A complementary Tailwind CSS plugin",
   "main": "dist/index.cjs",
   "types": "dist/index.d.ts",
+  "module": "dist/headlessui.esm.js",
   "license": "MIT",
   "files": [
     "README.md",
     "dist"
   ],
   "exports": {
-    "require": "./dist/index.cjs"
+    "import": "./dist/headlessui.esm.js",
+    "require": "./dist/index.cjs",
+    "types": "./dist/index.d.ts"
   },
   "sideEffects": false,
   "engines": {

